src/pages/PageProduct/PageProduct.tsx

```diff
@@ -0,0 +1,114 @@
+import { ApiProduct } from '../../api/Api';
+import { Grogu } from '../../components/Grogu/Grogu';
+import DeleteLogo from '/assets/images/delete.png';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+import {
+  IdataCardOneProduct,
+  IdataProduct,
+} from '../../type/interfaces.interface';
+import style from './_pageProduct.module.scss';
+
+export function PageProduct(): JSX.Element {
+  const [groguSpinner, setGroguSpinner] = useState<boolean>(true);
+  const [dataCard, setDataCard] = useState<IdataProduct>();
+  const [card, setCard] = useState<IdataCardOneProduct>();
+  const location = useLocation();
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const cardRef = useRef<HTMLDivElement>(null);
+
+  const handleClickOutside = useCallback(
```

Comment:
this logic can be moved to custom hook and reused
useEffect callback can return a clean up function where you can unsubscribe from document click event